packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -526,16 +526,31 @@ export default function useAutocomplete(props) {
     // eslint-disable-next-line react-hooks/rules-of-hooks
     React.useEffect(() => {
       if (!inputRef.current || inputRef.current.nodeName !== 'INPUT') {
-        console.error(
-          [
-            `MUI: Unable to find the input element. It was resolved to ${inputRef.current} while an HTMLInputElement was expected.`,
-            `Instead, ${componentName} expects an input element.`,
-            '',
-            componentName === 'useAutocomplete'
-              ? 'Make sure you have binded getInputProps correctly and that the normal ref/effect resolutions order is guaranteed.'
-              : 'Make sure you have customized the input component correctly.',
-          ].join('\n'),
-        );
+        if (
+          inputRef.current &&
+          inputRef.current.nodeName === 'TEXTAREA' &&
+          componentName === 'Autocomplete'
```

Comment:
Have you found it necessary to do this check? I think it shouldn't matter what component uses `useAutocomplete` - the behavior should be the same.

Reply:
removed the check